lib/Sema/CSSimplify.cpp

```diff
@@ -1501,6 +1501,10 @@ shouldOpenExistentialCallArgument(
     adjustments |= OpenedExistentialAdjustmentFlags::InOut;
   }
 
+  // The argument may be a "var" instead of a "let".
+  if (auto lv = dyn_cast<LValueType>(argTy->getCanonicalType()))
+    argTy = lv->getObjectType();
```

Comment:
A better check would be `argTy->is<LValueType>()` and you'd need to add a new `OpenedExistentialAdjustmentFlags::LValue` and re-add that later just like `Inout` does.